pkg/virt-controller/watch/vm.go

```diff
@@ -2646,6 +2645,13 @@ func (c *VMController) sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachin
 				}
 			}
 		}
+
+		if c.clusterConfig.HotplugNetworkInterfacesEnabled() {
```

Comment:
Why this is not under the `syncErr == nil` condition?
I would not expect it to be patched if we have a `syncErr != nil` condition.

Reply:
If you fail to update the vm you won't get the MAC from kmp. So it is wrong to patch the vmi in this case.